pkg/controller/intrusiondetection/intrusiondetection_controller.go

```diff
@@ -247,12 +254,29 @@ func (r *ReconcileIntrusionDetection) Reconcile(ctx context.Context, request rec
 	r.status.OnCRFound()
 	reqLogger.V(2).Info("Loaded config", "config", instance)
 
-	if err := r.setDefaultsOnIntrusionDetection(ctx, instance); err != nil {
+	managementClusterConnection, err := utils.GetManagementClusterConnection(ctx, r.client)
+	if err != nil {
+		log.Error(err, "Failed to read ManagementClusterConnection")
+		r.status.SetDegraded("Failed to read ManagementClusterConnection", err.Error())
```

Comment:
This needs to use one of the `TigeraStatusReason` values as the first argument.

Reply:
It should also use the `r.SetDegraded` instead of directly using the `r.status.SetDegraded`